### ISSUE TITLE
[Feature request] Allow filter branches

### ISSUE THREAD

**Issue by Aokromes** [useful-forks/useful-forks.github.io#38]
how about allow search only selected branches?
on that way we can filter out unwanted branches.

**Comment by Faq** [comment]
But then You should know precise name of it or meant is some other criteria?

**Comment by Aokromes** [comment]
yes, filter by master branch or dev branch or whatever branch :)

**Comment by payne911** [comment]
As it is, there is no branch scanning: the only branch I end up looking at is the "`default_branch`" which ends up being either `master`, `main`, or (I think) whatever the main branch is changed in the repo's configuration by the owner.

The reason for that is the amount of API calls that would be required to scan each branch individually would be restrictive.

Thus, a "negative filter" that would allow people to filter by branch would be rather useless. However, what could be discussed is the idea of a "positive filter" which would allow people to input which branches to compare (thus replace the value of the "`default_branch`"). But to be honest, that seems like a low-value feature. I do not personally think I'll spend time on that.

**Comment by lanrat** [comment]
For another data-point, I too would be interested in having the ability to search non-default branches. This would be useful for uncovering feature branches for projects that have yet yet to be merged into a default branch.

By default I'd like to be able to search all branches with activity, but I am aware that this may be infeasible due to the API limits.

One possible alternative or way to accomplish this could be to poll the GitHub repository Insights Network Graph endpoints. That seems to do something similar to this tool, but does return information on every possible branch.

Example: https://github.com/useful-forks/useful-forks.github.io/network

**Comment by esauvisky** [comment]
> For another data-point, I too would be interested in having the ability to search non-default branches. This would be useful for uncovering feature branches for projects that have yet yet to be merged into a default branch.
> 
> By default I'd like to be able to search all branches with activity, but I am aware that this may be infeasible due to the API limits.
> 
> One possible alternative or way to accomplish this could be to poll the GitHub repository Insights Network Graph endpoints. That seems to do something similar to this tool, but does return information on every possible branch.
> 
> Example: https://github.com/useful-forks/useful-forks.github.io/network

I second this. I ended up here by looking for a similar functionality which was being able to filter or at least view forks' commits (particularly messages) in a more data friendly format, given the Insights Network Graph shows them while hovering each commit (and **does not** make any extra API calls when doing it so).

The reasoning behind it is that very often when I'm looking at the network graph I find myself having to go through all forks' commits one by one to see what the changes were about until I find something related to what I am looking for, therefore having a more data friendly visualization of the same data would be very useful.

I understand that doing this with individual API calls is not realistic, but figuring out a way of leveraging the insights network graph endpoint's data would definitely open up new possibilities for more features.
